### PR TITLE
Add directory container archiver

### LINKED ISSUE #46: Fix case on windows env and figure out whether there is any way to do CI integration under windows env
Test under sofa-ark-springboot-starter module, for example: `ArkBootRunnerTest` will fail under windows env because of `RunnerHelper` for `prepareForTest.sh` 

And we can try to figure out whether there is any way  to do CI integration under windows env

### PR DESCRIPTION
**Pull request by QilongZhang**
fix #46

### CODE REVIEW

File: sofa-ark-parent/core-impl/archive/src/main/java/com/alipay/sofa/ark/bootstrap/ClasspathLauncher.java [hunk 1]
```diff
@@ -169,6 +183,108 @@ public InputStream getInputStream(ZipEntry zipEntry) throws IOException {
...
+                            || url.getPath().contains(
+                                String.format("%ssofa-ark-parent%score%scommon%starget%sclasses",
+                                    File.separator, File.separator, File.separator, File.separator,
+                                    File.separator))
+                            || url.getPath().contains(
+                                String.format(
+                                    "%ssofa-ark-parent%score%sexception%starget%sclasses",
+                                    File.separator, File.separator, File.separator, File.separator,
+                                    File.separator))) {
+                            containerClassPath.add(url);
```
Comment (straybirdzls) [inline sofa-ark-parent/core-impl/archive/src/main/java/com/alipay/sofa/ark/bootstrap/ClasspathLauncher.java]: These codes could be more elegant combined with  "ark.lib.properties" , currently it's a little bit hard to read them

File: sofa-ark-parent/support/ark-support-starter/src/test/java/com/alipay/sofa/ark/support/ArkJUnit4RunnerTest.java [hunk 2]
```diff
@@ -14,18 +14,18 @@
...
-import com.alipay.sofa.ark.runner.TestJUnit4Runner;
+import com.alipay.sofa.ark.support.runner.ArkJUnit4Runner;
 import org.junit.*;
 import org.junit.runner.RunWith;
 
 /**
  * @author qilong.zql 18/3/29-下午1:47
  */
-@RunWith(TestJUnit4Runner.class)
+@RunWith(ArkJUnit4Runner.class)
```
Comment (straybirdzls) [inline sofa-ark-parent/support/ark-support-starter/src/test/java/com/alipay/sofa/ark/support/ArkJUnit4RunnerTest.java]: This file header has chinese 😄

**Comment by straybirdzls** [comment]
@QilongZhang 

There is a big decrease on code coverage, please  have a check

image

File: sofa-ark-parent/core-impl/archive/src/main/java/com/alipay/sofa/ark/bootstrap/ClasspathLauncher.java [hunk 3]
```diff
@@ -169,6 +183,61 @@ public InputStream getInputStream(ZipEntry zipEntry) throws IOException {
...
+            for (String marker : arkContainerJarMarkers) {
+                for (URL url : urls) {
+                    if (url.getPath().contains(marker)) {
+                        containerClassPath.add(url);
+                    }
+                }
+            }
+
+            return arkContainerJarMarkers.size() != containerClassPath.size() ? null
+                : containerClassPath.toArray(new URL[] {});
```
[the remark this answers is not in the document]
Comment (QilongZhang) [reply sofa-ark-parent/core-impl/archive/src/main/java/com/alipay/sofa/ark/bootstrap/ClasspathLauncher.java]: so pity... Thanks for your troubleshoot, i'll fix it later

File: .travis.yml [hunk 4]
```diff
@@ -10,6 +10,10 @@ install:
...
 script:
 - sh ./check_format.sh
+- cd sofa-ark-samples && mvn clean install
+- cd sample-ark-plugin && mvn clean install
+- cd ../sample-springboot-ark && mvn test
+- cd ../..
 
 after_success:
-- mvn clean test cobertura:cobertura coveralls:report
+- mvn clean test co bertura:cobertura coveralls:report
```
Comment (straybirdzls) [inline .travis.yml]: why change here?

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# Do not build on tags
+skip_tags: true
+
+# build and install artifacts
+build_script:
+  - mvn clean install -DskipTests
+
+# verify artifacts
+test_script:
+  - mvn test
```
Comment (straybirdzls) [inline appveyor.yml]: Should we also verify samples on windows?

**Pull request merged**